server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java

```diff
@@ -213,12 +216,13 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
         throw new UnsupportedOperationException("Parsing is implemented in parse(), this method should NEVER be called");
     }
 
-    public static class GeoPointFieldType extends MappedFieldType {
+    public static class GeoPointFieldType extends AbstractGeometryFieldMapper.AbstractGeometryFieldType<Geometry, Geometry> {
         public GeoPointFieldType() {
         }
 
         GeoPointFieldType(GeoPointFieldType ref) {
             super(ref);
+            this.geometryQueryBuilder  = ref.geometryQueryBuilder;
```

Comment:
That feels a bit awkward. I think that should go into ` AbstractGeometryFieldMapper.AbstractGeometryFieldType` if we need to clone it. I think the reason it is not cloned there already is that we override it always in setupFileds anyway. But cloning it in `AbstractGeometryFieldType` seems like a logical thing to do here.

Reply:
OK, on it

Reply:
still working on the refactor, if #53099 is a better alternative I'm fine to go with that

Reply:
@imotov @iverase I'm putting the refactor on hold pending outcome of review on #53099 which looks preferable to me, for the following reason: it looks like I'd also have to add SearchableGeometryQueryBuilder, which makes the Geo related classes in `index.query` diverge further from how the rest of the queries are built

Reply:
Refactored with help from a patch from @iversase thanks!